packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx

```diff
@@ -113,6 +114,103 @@ const maybeZoomAction = (e?: MouseEvent | null) => e != null && !e.ctrlKey && !e
 const getDataLinksFallback: GetDataLinksCallback = () => [];
 const getAdHocFiltersFallback: GetAdHocFiltersCallback = () => [];
 
+/**
+ * Enables panning the time range by dragging x-axis labels.
+ * Provides visual feedback (grab/grabbing cursor) and real-time grid updates during drag.
+ * Calls queryZoom() on drag end to update dashboard time range if drag exceeds MIN_ZOOM_DIST.
+ * Mutates u.scales.x.range temporarily during drag.
+ * @internal - exported for testing only
+ */
+export const setupXAxisPan = (u: uPlot, queryZoom: (range: { from: number; to: number }) => void) => {
+  let xAxes = u.root.querySelectorAll('.u-axis');
+  let xAxis = xAxes[0];
+
+  if (!xAxis || !(xAxis instanceof HTMLElement)) {
+    return;
+  }
+
+  const xAxisEl = xAxis;
+
+  xAxisEl.addEventListener('mouseenter', () => {
+    xAxisEl.style.cursor = 'grab';
+  });
+
+  xAxisEl.addEventListener('mouseleave', () => {
+    xAxisEl.style.cursor = '';
+  });
+
+  xAxisEl.addEventListener('mousedown', (e: Event) => {
+    if (!(e instanceof MouseEvent)) {
+      return;
+    }
+    e.preventDefault();
+
+    xAxisEl.style.cursor = 'grabbing';
+
+    let xScale = u.scales.x;
+    let originalRange = xScale.range;
+
+    let rect = u.over.getBoundingClientRect();
+    let startX = e.clientX - rect.left;
+    let startMin = xScale.min!;
+    let startMax = xScale.max!;
+    let unitsPerPx = (startMax - startMin) / (u.bbox.width / uPlot.pxRatio);
+
+    let isDragging = false;
+    let dragMin = startMin;
+    let dragMax = startMax;
+
+    xScale.range = (u, initMin, initMax, scaleKey) => {
+      if (isDragging) {
+        return [dragMin, dragMax];
+      }
+      return typeof originalRange === 'function' ? originalRange(u, initMin, initMax, scaleKey) : [initMin, initMax];
+    };
```

Comment:
let's see if we can do this differently without mutating the instance.

Reply:
I liked the results of the alternative approach so much, I just cherry picked those commits over to this branch.